huaweicloud/services/apig/resource_huaweicloud_apig_signature_associate.go

```diff
@@ -0,0 +1,303 @@
+		// for a short period of time, and the polling is performed by incrementing the time here.
+		MinTimeout: 2 * time.Second,
+	}
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmt.Errorf("error waiting for the binding completed: %s", err)
+	}
+	return nil
+}
+
+func resourceSignatureAssociateCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	client, err := cfg.ApigV2Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating APIG v2 client: %s", err)
+	}
+	var (
+		instanceId = d.Get("instance_id").(string)
+		signId     = d.Get("signature_id").(string)
+		publishIds = d.Get("publish_ids").(*schema.Set)
+
+		opts = signs.BindOpts{
+			InstanceId:  instanceId,
+			SignatureId: signId,
+			PublishIds:  utils.ExpandToStringListBySet(publishIds),
+		}
+	)
+	err = bindSignatureToApis(ctx, client, opts, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		diag.FromErr(err)
+	}
+	d.SetId(fmt.Sprintf("%s/%s", instanceId, signId))
+
+	return resourceSignatureAssociateRead(ctx, d, meta)
+}
+
+func buildSignBindApiListOpts(instanceId, signId string) signs.ListBindOpts {
+	return signs.ListBindOpts{
+		InstanceId:  instanceId,
+		SignatureId: signId,
+		Limit:       500,
+	}
+}
+
+func flattenApiPublishIdsForSignature(apiList []signs.SignBindApiInfo) []string {
+	if len(apiList) < 1 {
+		return nil
+	}
+
+	result := make([]string, len(apiList))
+	for i, val := range apiList {
+		result[i] = val.PublishId
+	}
+	return result
+}
+
+func resourceSignatureAssociateRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.ApigV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating APIG v2 client: %s", err)
+	}
+
+	var (
+		instanceId = d.Get("instance_id").(string)
+		signId     = d.Get("signature_id").(string)
+		opts       = buildSignBindApiListOpts(instanceId, signId)
+	)
+
+	resp, err := signs.ListBind(client, opts)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "Signature association")
+	}
+	if len(resp) < 1 {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
+	}
+
+	return diag.FromErr(d.Set("publish_ids", flattenApiPublishIdsForSignature(resp)))
+}
+
+func signatureUnbindingRefreshFunc(client *golangsdk.ServiceClient, instanceId, signId, bandId string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		opts := buildSignBindApiListOpts(instanceId, signId)
+		resp, err := signs.ListBind(client, opts)
+		if err != nil {
+			return resp, "", err
+		}
+		for _, val := range resp {
+			if val.BindId == bandId {
+				return resp, "PENDING", nil
+			}
+		}
+		return resp, "COMPLETED", nil
+	}
+}
+
+func unbindSignatureFromApis(ctx context.Context, client *golangsdk.ServiceClient, instanceId, signId string,
```

Comment:
it seems that this function will unbind all of the publish ids, please check it.

Reply:
done.